### PR TITLE
F/wpc 538 add call to action string

### PR DESCRIPTION
**Pull request by lakeside763**
Test: Editor: https://editor.fwpc538ad.wepublish.dev
PR: As a publisher, i don't want to be able to enter richtext for the CTA Text
https://wepublish.atlassian.net/browse/WPC-538
 
Replace callToActionText with callToActionString and keep the Richtext block data for old version compatibility

callToActionString
.

### CODE REVIEW

File: packages/api/package.json [hunk 1]
```diff
@@ -79,7 +79,6 @@
     "apollo-server": "^2.17.0",
     "apollo-server-testing": "^2.17.0",
     "dotenv": "^8.2.0",
-    "jest": "^26.4.2",
```
Comment (bishoymelek) [inline packages/api/package.json:82]: I can't get it :) why did you remove jest?

**Review by camilleory — APPROVED** [review]

**Review by bishoymelek — DISMISSED** [review]
🎉

**Review by penk85 — DISMISSED** [review]

File: packages/api-db-mongodb/src/migration.ts [hunk 2]
```diff
@@ -641,6 +642,27 @@ export const Migrations: Migration[] = [
...
+        callToActionText: {$exists: true}
+      })
+      if (peerProfile) {
+        await db.collection(CollectionName.PeerProfiles).updateOne(
+          {
+            _id: peerProfile._id
+          },
+          {
+            $set: {
+              callToActionString: richTextToString('', peerProfile?.callToActionText)
```
Comment (roosnic1) [inline packages/api-db-mongodb/src/migration.ts:660]: the `?` is not necessary if you check above if `peerProfile` exists.

File: packages/api/src/graphql/peer.ts [hunk 3]
```diff
@@ -49,7 +49,8 @@ export const GraphQLPeerProfile = new GraphQLObjectType<PeerProfile, Context>({
     },
     hostURL: {type: GraphQLNonNull(GraphQLString)},
     websiteURL: {type: GraphQLNonNull(GraphQLString)},
-    callToActionText: {type: GraphQLNonNull(GraphQLRichText)},
+    callToActionText: {type: GraphQLRichText},
```
Comment (roosnic1) [inline packages/api/src/graphql/peer.ts:52]: why did you remove the `GraphQlNonNull` from `callToActionText`?

File: packages/editor/src/client/api/peer.graphql [hunk 4]
```diff
@@ -7,6 +7,7 @@ fragment FullPeerProfile on PeerProfile {
     ...ImageRef
   }
   callToActionText
```
Comment (roosnic1) [inline packages/editor/src/client/api/peer.graphql:9]: in the editor you can remove `callToActionText` there it's not needed anymore

**Review by roosnic1 — CHANGES_REQUESTED** [review]
Looks good but still needs some changes.

File: examples/website/src/shared/route/articleTemplateContainer.tsx [hunk 5]
```diff
@@ -422,7 +422,11 @@ export function PeerProfileBlock({peer, article}: PeerProfileBlockProps) {
         <div className={css(PeerProfileFiller)} />
       </div>
       <div className={css(PeerProfileCallToActionURL)}>
-        {peer?.callToActionText?.length && (
+        {peer?.callToActionString ? (
+          <a target="_blank" rel="noreferrer" href={peer?.callToActionURL}>
+            {peer?.callToActionString}
+          </a>
+        ) : (
           <a target="_blank" rel="noreferrer" href={peer?.callToActionURL}>
```
Comment (penk85) [inline examples/website/src/shared/route/articleTemplateContainer.tsx:430]: Wouldn't the `callToActionText` also be possible to be empty here? Shouldn't the `callToActionText` and `callToActionString` both be rendered conditionally, independently?

File: packages/api-db-mongodb/src/migration.ts [hunk 6]
```diff
@@ -644,7 +645,6 @@ export const Migrations: Migration[] = [
     }
   },
   {
-    // add slug to existing paymentMethods
```
Comment (penk85) [inline packages/api-db-mongodb/src/migration.ts:647]: Why did you remove this line?

**Pull request closed without merging**